include/dd/StochasticNoiseOperationTable.hpp

```diff
@@ -41,12 +29,12 @@ namespace dd {
             table.resize(nvars);
         }
 
-        void insert(NoiseOperationKind kind, Qubit target, const Edge& r) {
+        void insert(std::uint_fast8_t kind, Qubit target, const Edge& r) {
```

Comment:
The enum is defined in the QFR and therefore not available here. I thought about moving the StochasticNoiseOperationTable.hpp into the simulator, but this is a bad idea since garbage collection is done within the ddPackage. Copying the enum from the QFR is ugly because of code duplication. We talked in discord about moving the enum from the QFR into the ddPackage, which you did not like. 

Interpreting the enum as a std::uint_fast8_t seems like a convenient workaround.

Reply:
This still does not seem like the best solution to me. These noise operations are very different conceptually compared to the `StandardOperations` in the QFR. If I remember correctly, `ATrue` does not even represent a unitary operation.
So, in my opinion, it would definitely make sense to create a separate `NoiseOperation` enum. (Even if it shares the `I, X, Y, Z` operations with the `OpType`.

I do see the point, that moving the Table outside of the Package does not really make sense.

However, the package should still be self-contained and not depend on external enums to be defined and consistent.

Reply:
I extended the functionality of the stochasticNoiseOperation Cache to include all Operations defined in `OpType`. Conceptionally it does not make much of a difference but significantly improves the performance. Copying the `OpType` enum, therefore, entails duplicating the entire enum and would still require keeping the enums in sync.

Reply:
I'll keep the discussion going in the other thread.